comp110-worksheet-6/OxoBoard.cs

```diff
@@ -8,69 +8,380 @@ namespace comp110_worksheet_6
+            grids = new Grid[width, height];
+            this.inARow = inARow;
+
+            PopulateGrid();
+            horizontalASCII = GetHorizontalString();
 		}
 
-		// Return the contents of the specified square.
-		public Mark GetSquare(int x, int y)
+        /// <summary>
+        /// Populates the multidimensional array storing the grids class
+        /// and instantiates a new class and assigns the appropriate variables
+        /// </summary>
+        private void PopulateGrid()
+        {
+            for (int x = 0; x < grids.GetLength(0); x++)
+            {
+                for (int y = 0; y < grids.GetLength(1); y++)
+                {
+                    grids[x, y] = new Grid(Mark.None, new Tuple<int, int>(x, y));
+                    AssignWinningPositions(grids[x, y]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Used to cycle through each Grid's possible combinations and then cycles through
+        /// those combinations to see if they all match one specific mark
+        /// </summary>
+        /// <param name="gridTile"></param>
+        /// <returns>
+        /// If a combination has been met then it returns the mark that won otherwise it
+        /// returns none which states no one has one yet
+        /// </returns>
+        private Mark CheckGridTile(Grid gridTile)
+        {
+            List<Mark> marks;
+
+            foreach (var winningPositions in gridTile.winPositions)
+            {
+                marks = new List<Mark>();
+                for (int i = 0; i < winningPositions.Count; i++)
+                { 
+                    if (winningPositions[i].Item1 < grids.GetLength(0) && winningPositions[i].Item2 < grids.GetLength(1))
+                    {
+                        marks.Add(GetSquare(winningPositions[i].Item1, winningPositions[i].Item2));
+                    }
+                    
+                }
+                
+                if (IsAllTheSameMarks(marks.ToArray()) && marks.Count == inARow)
+                {
+                    Console.WriteLine(marks[0] + " won!");
+                    return marks[0];
+                }
+            }
+
+            return Mark.None;
+        }
+
+        /// <summary>
+        /// Acts like a duplicate checker for the marks in an array
+        /// </summary>
+        /// <param name="marks"></param>
+        /// <returns>
+        /// Returns true if all of the marks are the same, but false if 
+        /// there is a duplicate of it, or if any contain none
+        /// </returns>
+        private bool IsAllTheSameMarks(Mark[] marks)
+        {
+            for (int i = 0; i < marks.Length; i++)
+            {
+                for (int j = 0; j < marks.Length; j++)
+                {
+                    if (marks[i] != marks[j] || marks[j] == Mark.None || marks[i] == Mark.None)
+                    {
+                        return false;
+                    }
+                }
+            }
```

Comment:
This is O(n^2) -- it could be made O(n) by just checking each element against the one before it (checking every possible pair is overkill for just checking all elements are the same)